join.c

```diff
@@ -125,9 +131,9 @@ void joinIdleCallback(Connection *conn)
     time(&now);
 
     if (difftime(now, state->start) > rr->config->join_timeout) {
-        LOG_ERROR("timed out trying to join cluster");
+        LOG_ERROR("Cluster join: timed out, took longer than %d seconds", rr->config->join_timeout);
```

Comment:
what's the practical difference?  it hit the limit, so if the limit is 60s and we only see it after 61s because of some hang, what difference does it make?

Reply:
I'd figure that the limit would be even better as it would give a user something to figure out if they don't know where that limit is coming from, vs a random time doesn't give them insight into whats going on

Reply:
You're right, unless something unexpected happens and it blocks for a very long time... although the log timestamp would also tell the story.

Reply:
unsure if you are happy with this.  note, I added another failure case now.  we can short circuit retries if we get an unrecoverable error.  (speeds up testing a lot, no more retry on the try to readd node with old node id test)